pkg/core/types.go

```diff
@@ -0,0 +1,7 @@
+package core
+
+import "github.com/spf13/cobra"
+
+type Plugin interface {
+	CreateRootCmd( /*TODO: Define config struct to add as arg here*/ ) *cobra.Command
```

Comment:
Benefit of wrapping cobra root CLI?

I would say we could do validate that but my preference is to do static linting rather than validating thing at runtime.
See #58 

It is hard to wrap my head around this approach without seeing config struct and since I do not see it here this probably means that you are still figuring it out.

Reply:
Another bummer here is that by forcing us to have `ROOT` we instantly telling people that we cannot have 2 root commands or set of the commands that people can bring into their CLI. etc.